File: ECommerce.View/src/app/core/models/order.ts
export enum OrderStatus {
  Pending = "Pending",
  Confirmed = "Confirmed",
  Processing = "Processing",
  Packed = "Packed",
  Shipped = "Shipped",
  Delivered = "Delivered",
  Cancelled = "Cancelled",
  PreOrder = "PreOrder",
  Hold = "Hold",
  Return = "Return",
  Exchange = "Exchange",
  ReturnProcess = "ReturnProcess",
  Refund = "Refund",
}

export interface OrderLog {
  id: number;
  statusFrom: string;
  statusTo: string;
  changedBy: string;
  note?: string;
  createdAt: string;
}

export interface OrderNote {
  id: number;
  adminName: string;
  content: string;
  createdAt: string;
}

export interface OrderItem {
  id?: number;
  productId: number;
  productName: string;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
  size?: string;
  imageUrl?: string;
  color?: string;
  isStockAvailable?: boolean;
}

export interface Order {
  id: number;
  orderNumber: string;
  customerName: string;
  customerPhone: string;
  shippingAddress: string;
  subTotal: number;
  tax: number;
  shippingCost: number;
  total: number;
  itemsCount: number;
  status: OrderStatus;
  createdAt: string;
  items: OrderItem[];

  deliveryMethodId?: number;
  discount?: number;
  advancePayment?: number;
  updatedAt?: string;
  paymentStatus?: string;
  city?: string;
  area?: string;
  isPreOrder?: boolean;
  isStockAvailable?: boolean;
  adminNote?: string;
  customerNote?: string;
  sourcePageId?: number;
  sourcePageName?: string;
  socialMediaSourceId?: number;
  socialMediaSourceName?: string;
  logs?: OrderLog[];
  notes?: OrderNote[];
}

export interface OrdersQueryParams {
  searchTerm: string;
  status: "All" | OrderStatus;
  dateRange: "Today" | "Yesterday" | "Last 7 Days" | "Last 30 Days" | "This Year" | "All Time" | "Custom";
  startDate?: string;
  endDate?: string;
  page: number;
  pageSize: number;
  preOrderOnly?: boolean;
  websiteOnly?: boolean;
  manualOnly?: boolean;
  sourcePageId?: number;
  socialMediaSourceId?: number;
  customerPhone?: string;
  productId?: number;
  orderNumber?: string;
}

export interface OrderStats {
  totalOrders: number;
  processing: number;
  totalRevenue: number;
  refundRequests: number;
}

File: ECommerce.Core/Entities/Order.cs
using System;
using System.Collections.Generic;
using ECommerce.Core.Enums;

namespace ECommerce.Core.Entities;

public class Order : BaseEntity
{
    public string OrderNumber { get; set; } = string.Empty;

    public string CustomerName { get; set; } = string.Empty;
    // NOTE: CustomerPhone is a flat string (no FK to Customer) by design.
    // Orders can be placed by non-registered users, so a FK is not enforced here.
    public string CustomerPhone { get; set; } = string.Empty;
    public string ShippingAddress { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string Area { get; set; } = string.Empty;

    public decimal SubTotal { get; set; }
    public decimal Tax { get; set; }
    public decimal ShippingCost { get; set; }
    public decimal Discount { get; set; }
    public decimal AdvancePayment { get; set; }
    public decimal Total { get; set; }

    public int? UpazilaId { get; set; }
    public Location.Upazila? Upazila { get; set; }

    public int? DivisionId { get; set; }
    public Location.Division? Division { get; set; }

    public int? DistrictId { get; set; }
    public Location.District? District { get; set; }

    public int? DeliveryMethodId { get; set; }
    public DeliveryMethod? DeliveryMethod { get; set; }

    public OrderStatus Status { get; set; } = OrderStatus.Pending;

    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();

    public string? CreatedIp { get; set; }
    public bool IsPreOrder { get; set; }
    public string? AdminNote { get; set; }
    public string? CustomerNote { get; set; }
    public int? SourcePageId { get; set; }
    public SourcePage? SourcePage { get; set; }

    public int? SocialMediaSourceId { get; set; }
    public SocialMediaSource? SocialMediaSource { get; set; }

    public string? SessionId { get; set; }
    public string? ReferrerUrl { get; set; }
    public string? UtmSource { get; set; }
    public string? UtmCampaign { get; set; }
    public string? UtmAdset { get; set; }
    public string? UtmAd { get; set; }
    public string? Fbclid { get; set; }
    public string? DeviceType { get; set; }
    public string? Browser { get; set; }

    public ICollection<OrderLog> Logs { get; set; } = new List<OrderLog>();
    public ICollection<OrderNote> Notes { get; set; } = new List<OrderNote>();
}
